executor/src/witgen/jit/processor.rs

```diff
@@ -60,15 +58,28 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> Processor<'a, T, FixedEv
             fixed_evaluator,
             identities,
             occurrences,
-            block_size,
-            check_block_shape,
+            block_size: 1,
+            check_block_shape: false,
             requested_known_vars: requested_known_vars.into_iter().collect(),
             max_branch_depth,
         }
     }
 
+    /// Sets the block size.
+    pub fn with_block_size(mut self, block_size: usize) -> Self {
+        self.block_size = block_size;
+        self
+    }
+
+    /// Activates the check to see if the code for two subsequently generated
+    /// blocks conflicts.
+    pub fn with_block_shape_check(mut self) -> Self {
+        self.check_block_shape = true;
+        self
+    }
+
     pub fn generate_code(
-        &self,
```

Comment:
This ensures that we cannot modify the parameters after the call to `generate_code`, which is the only public function that is not a builder function.